### ISSUE TITLE
Squel doesn't support the WINDOW clause

### ISSUE THREAD

**Issue by SGrondin** [hiddentao/squel#245]
The `WINDOW` clause goes in between `HAVING` and `ORDER BY`. It would be helpful to add support for it for those of us who need this advanced feature and can't use Squel at the moment. Thanks!

<img width="126" alt="screen shot 2016-06-10 at 8 09 13 pm" src="https://cloud.githubusercontent.com/assets/2441123/15981692/3fc89af4-2f47-11e6-9553-503ea7d7e8c0.png">

**Comment by SGrondin** [comment]
I've just found the [Custom Queries](https://hiddentao.github.io/squel/#custom_queries) chapter of the docs. I'm going to build my own SELECT with WINDOW support, but adding it to Squel's default SELECT Builder would be great. It's part of the official Postgres SQL dialect, so it feels like a clear omission in Squel :)

**Comment by hiddentao** [comment]
Squel has a postgres “flavour”. I’d be happy to accept a tested PR for that.


On 11 June 2016 at 08:43:59, Simon Grondin (notifications@github.com) wrote:

I've just found the Custom Queries chapter of the docs. I'm going to build my own SELECT with WINDOW support, but adding it to Squel's default SELECT Builder would be great. It's part of the official Postgres SQL dialect, so it feels like a clear omission in Squel :)

**Comment by SGrondin** [comment]
I'll submit a PR once I'm more familiar with Squel's internals and once I've added all the WINDOW features to my fork. My version already works, but it doesn't support everything that the Postgres standard offers.

**Comment by SGrondin** [comment]
I've got a basic question, not worth opening an Issue for.

@hiddentao I've looked into the source and I can't seem to find how to disable Parameters. Basically, I **don't** want Squel to replace my `?` placeholders. Other than setting `parameterCharacter` to a character that's guaranteed to not be in my Queries, is there a way to disable this feature? Thanks

**Comment by hiddentao** [comment]
Not really. Setting the parameterCharacter to a some other string is the way to do it. Hacky, I know.


On 11 June 2016 at 23:44:52, Simon Grondin (notifications@github.com) wrote:

I've got a basic question, not worth opening an Issue for.

@hiddentao I've looked into the source and I can't seem to find how to disable Parameters. Basically, I don't want Squel to replace my ? placeholders. Other than setting parameterCharacter to a character that's guaranteed to not be in my Queries, is there a way to disable this feature? Thanks

**Comment by SGrondin** [comment]
Thanks, I'll make a PR for that too (if it turns to not be too difficult). If it's a difficult task, is it the kind of feature that's on your roadmap?

**Comment by hiddentao** [comment]
The roadmap is open. If enough people deem something useful it gets added in. I'm hesitant to add anything the core query builders that's not in the SQL standard. But that's why we have "flavours", so that people can add engine-specific code.
